src/main/java/com/gooddata/executeafm/afm/MeasureDefinition.java

```diff
@@ -33,18 +40,34 @@ default String getUri() {
     }
 
     /**
-     * Returns the qualifier, qualifying the {@link com.gooddata.md.Metric}.
-     * @return qualifier of measure
+     * Returns all the qualifiers used by the measure definition and its encapsulated objects.
+     * <p>
+     * This information comes handy if it is necessary, for example, to convert the measure definition to use just the URI object qualifiers instead of the
+     * identifier object qualifiers. It can be used to gather these for a conversion service.
+     *
+     * @return all the qualifiers the measure definition uses, even in its encapsulated objects (apart from the measure filters)
      */
     @JsonIgnore
-    ObjQualifier getObjQualifier();
+    Collection<ObjQualifier> getObjQualifiers();
```

Comment:
please do not make backward incompatible changes on interfaces, if we are sure the "old" method is not usefull, make it deprecated and create the new one beside that.

Reply:
Fixed. I got the old methods back and used them as a proxy to the new implementation. There are tests for it as well.

The new measure definitions throw an exception when the methods are called.

I also updated the interface of the new methods and they use the `Function` method instead of the map now. It was a good suggestion as it cleared the code a bit thanks to the easier checks and validations because `Optional` API can be used now. Thanks 👍